pkg/controllers/resources/ingresses/syncer.go

```diff
@@ -117,3 +133,82 @@ func (s *ingressSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *syncc
 
 	return patcher.CreateVirtualObject(ctx, event.Host, vIngress, s.EventRecorder(), true)
 }
+
+func (s *ingressSyncer) ExcludeVirtual(obj client.Object) bool {
+	ingress, ok := obj.(*networkingv1.Ingress)
+	if !ok ||
+		((ingress.Spec.IngressClassName == nil || *ingress.Spec.IngressClassName == "") && selector.IsLabelSelectorEmpty(s.labelSelector)) {
+		return false
+	}
+
+	ingressClass := &networkingv1.IngressClass{}
+	err := s.physicalClusterClient.Get(context.Background(), types.NamespacedName{Name: *ingress.Spec.IngressClassName}, ingressClass)
```

Comment:
This is a problem, we don't have permissions to do that when only ingress syncing is enabled